examples/fungible-token/src/storage_manager.rs

```diff
@@ -0,0 +1,94 @@
+use super::*;
+use near_sdk::json_types::{ValidAccountId, U128};
+use near_sdk::serde::Serialize;
+
+/// Price per 1 byte of storage from mainnet config after `0.18` release and protocol version `42`.
+/// It's 10 times lower than the genesis price.
+pub const STORAGE_PRICE_PER_BYTE: Balance = 10_000_000_000_000_000_000;
+
+#[derive(Serialize)]
+#[serde(crate = "near_sdk::serde")]
+pub struct AccountStorageBalance {
+    total: U128,
+    available: U128,
+}
+
+pub trait StorageManager {
+    fn storage_deposit(&mut self, account_id: Option<ValidAccountId>) -> AccountStorageBalance;
+
+    fn storage_withdraw(&mut self, amount: U128) -> AccountStorageBalance;
+
+    fn storage_minimum_balance(&self) -> U128;
+
+    fn storage_balance_of(&self, account_id: ValidAccountId) -> AccountStorageBalance;
+}
+
+#[near_bindgen]
+impl StorageManager for Contract {
+    #[payable]
+    fn storage_deposit(&mut self, account_id: Option<ValidAccountId>) -> AccountStorageBalance {
+        let amount = env::attached_deposit();
+        assert_eq!(
+            amount,
+            self.storage_minimum_balance().0,
+            "Requires attached deposit of the exact storage minimum balance"
```

Comment:
Something to call out is that this `storage_deposit()` implementation is specific to this FT use case, i.e., requiring exact amount that matches `storage_minimum_balance()`. The business logic depends on the contract's context and use case.